protocol-units/bridge/contracts/test/AtomicBridgeInitiator.t.sol

```diff
@@ -0,0 +1,208 @@
+        ) = atomicBridgeInitiator.bridgeTransfers(bridgeTransferId);
+
+        assertFalse(transferCompleted);
+        assertEq(transferAmount, wethAmount);
+        assertEq(transferOriginator, originator);
+        assertEq(transferRecipient, recipient);
+        assertEq(transferHashLock, hashLock);
+        assertGt(transferTimeLock, block.timestamp);
+
+        vm.stopPrank();
+    }
+
+    function testInitiateBridgeTransferWithEthAndWeth() public {
+        uint256 wethAmount = 1 ether;
+        uint256 ethAmount = 2 ether;
+        uint256 totalAmount = wethAmount + ethAmount;
+
+        // Ensure the originator has sufficient ETH
+        vm.deal(originator, 100 ether);
+
+        vm.startPrank(originator);
+        // Ensure WETH contract is correctly funded and transfer WETH to originator
+        weth.deposit{value: wethAmount}();
+
+        assertEq(weth.balanceOf(originator), wethAmount, "WETH balance mismatch");
+        vm.expectRevert();
+        // Try to initiate bridge transfer
+        atomicBridgeInitiator.initiateBridgeTransfer{value: ethAmount}(wethAmount, recipient, hashLock, timeLock);
+        weth.approve(address(atomicBridgeInitiator), wethAmount);
+        // Try to initiate bridge transfer
+        bytes32 bridgeTransferId =
+            atomicBridgeInitiator.initiateBridgeTransfer{value: ethAmount}(wethAmount, recipient, hashLock, timeLock);
+
+        // Fetch the details of the initiated bridge transfer
+        (
+            uint256 transferAmount,
+            address transferOriginator,
+            bytes32 transferRecipient,
+            bytes32 transferHashLock,
+            uint256 transferTimeLock,
+            bool completed
+        ) = atomicBridgeInitiator.bridgeTransfers(bridgeTransferId);
+
+        // Assertions
+        assertFalse(completed, "Bridge transfer not completed");
+        assertEq(transferAmount, totalAmount, "Transfer amount mismatch");
+        assertEq(transferOriginator, originator, "Originator address mismatch");
+        assertEq(transferRecipient, recipient, "Recipient address mismatch");
+        assertEq(transferHashLock, hashLock, "HashLock mismatch");
+        assertGt(transferTimeLock, block.timestamp, "TimeLock is not greater than current block timestamp");
+
+        vm.stopPrank();
+    }
+
+    function testRefundBridgeTransfer() public {
+        vm.deal(originator, 1 ether);
+        vm.startPrank(originator);
+
+        bytes32 bridgeTransferId = atomicBridgeInitiator.initiateBridgeTransfer{value: amount}(
+            0, // _wethAmount is 0
+            recipient,
+            hashLock,
+            timeLock
+        );
+
+        vm.stopPrank();
+
+        vm.warp(block.timestamp + timeLock + 1);
+        vm.startPrank(originator);
+        atomicBridgeInitiator.refundBridgeTransfer(bridgeTransferId);
+
+        (,,,,, bool completed) = atomicBridgeInitiator.bridgeTransfers(bridgeTransferId);
+        assertTrue(completed);
+
+        vm.stopPrank();
+    }
+}
```

Comment:
Please add a test showing how to iterate over the stored bridge transfers from your map by adding 10 or 20 of them. We should be able to iterate through all of these entries to assess functionality requirements for the bridge.

Ideally, to restore the atomic bridge state, remove entries once they are completed to streamline the state--this makes the restore of the system lighter.

If you want to "store" all bridge transfers, could you please provide the rationale why?
Also, if valid, may I ask you to consider using two separate maps to distinguish between completed and incomplete transfers? I need quick access to all incomplete items to restore blockchain state.

Reply:
As discussed, we will not be going with the strategy for indexing historical data. This would be done by indexing historical data within a blockheight range and listening for those events.